crates/bin/prove_block/src/main.rs

```diff
@@ -189,20 +193,62 @@ async fn main() -> Result<(), Box<dyn Error>> {
             }
         };
 
-    let state_update =
-        match provider.get_state_update(BlockId::Number(block_number)).await.expect("Failed to get state update") {
-            MaybePendingStateUpdate::Update(update) => update,
-            MaybePendingStateUpdate::PendingUpdate(_) => {
-                panic!("Block is still pending!")
+    let state_update = match provider.get_state_update(block_id).await.expect("Failed to get state update") {
+        MaybePendingStateUpdate::Update(update) => update,
+        MaybePendingStateUpdate::PendingUpdate(_) => {
+            panic!("Block is still pending!")
```

Comment:
dunno, I didn't write that ;) I'm guessing because we don't support a pending block right now (or intend to). I suspect that properly supporting this would require additional logic for retrying the block until it's no longer pending, etc. and that the caller could handle this detail more gracefully anyway.

A pending block usually means that consensus hasn't been reached for it, but I'm not sure precisely what it means in this context.

My suggestion would be to return an error more gracefully, but I'm not sure how to even produce this case anyway :shrug: